### DISCUSSION TITLE
v19.4.20

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#2767] [Releases]
## Overview
This is mainly a bugfix release with a couple of optimizations.
Most importantly we've fixed bugs regarding:
-  fixer.io currency exchange rate provider
- CSS merge
- indexes

Upgrading is highly suggested, but always backup and test before doing it.

## What's Changed
* Set php version for phpstan by @sreichel in https://github.com/OpenMage/magento-lts/pull/2692
* Do not autoload captcha class when disabled by @luigifab in https://github.com/OpenMage/magento-lts/pull/2681
* Do not crash when shipment does not exist by @luigifab in https://github.com/OpenMage/magento-lts/pull/2683
* Trimmed files by @luigifab in https://github.com/OpenMage/magento-lts/pull/2698
* Reduce again getId calls by @luigifab in https://github.com/OpenMage/magento-lts/pull/2699
* Add link to the product page and float default qty for bundle items by @luigifab in https://github.com/OpenMage/magento-lts/pull/2701
* Remove obsolete ACL resources from DB by @sreichel in https://github.com/OpenMage/magento-lts/pull/2706
* Allow labeler workflow to fail by @sreichel in https://github.com/OpenMage/magento-lts/pull/2710
* Bugfix to make exchange rate data with fixer.io work again by @dbachmann in https://github.com/OpenMage/magento-lts/pull/2694
* Fixes typo in join alias in indexer by @rubanooo in https://github.com/OpenMage/magento-lts/pull/2711
* Removed support for eAccelerator Cache Backend by @fballiano in https://github.com/OpenMage/magento-lts/pull/2712
* Set the right menu for reviews by @luigifab in https://github.com/OpenMage/magento-lts/pull/2680
* Version bump for next release by @fballiano in https://github.com/OpenMage/magento-lts/pull/2714
* Added confirmation before deleting website/store/storeview by @fballiano in https://github.com/OpenMage/magento-lts/pull/2717
* Fixed typo in copyright docblock by @fballiano in https://github.com/OpenMage/magento-lts/pull/2740
* Add method code in payment method list by @luigifab in https://github.com/OpenMage/magento-lts/pull/2735
* Set width:auto for td.massaction by @luigifab in https://github.com/OpenMage/magento-lts/pull/2718
* Fix data.title replace when it's null/undefined by @luigifab in https://github.com/OpenMage/magento-lts/pull/2719
* Add PHPCodeSniffer to workflow by @sreichel in https://github.com/OpenMage/magento-lts/pull/2708
* Use getStoreConfigFlag() instead of (bool)getStoreConfig() by @sreichel in https://github.com/OpenMage/magento-lts/pull/2747
* Replaces full class name with self by @sreichel in https://github.com/OpenMage/magento-lts/pull/2749
* Updated phpdocs: return $this by @sreichel in https://github.com/OpenMage/magento-lts/pull/2751
* Backport: #1149 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2745
* Fixed incorrect docblock for setLastRealOrderId() and getLastRealOrde… by @kiatng in https://github.com/OpenMage/magento-lts/pull/2752
* Fix: merge CSS files /w missing file by @sreichel in https://github.com/OpenMage/magento-lts/pull/2754
* Backport: #2315 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2746
* Fixed setting of source_model when adding new attribute and for multiselect. by @kiatng in https://github.com/OpenMage/magento-lts/pull/1293
* Remove declared properties accessed by magic getter in Paypal Config by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/2759
* Fix Order comment REST endpoint route param by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/2750
* Moved DDEV docs by @sreichel in https://github.com/OpenMage/magento-lts/pull/2764
* Add php-cs-fixer & PHPCompatibility check to workflow by @sreichel in https://github.com/OpenMage/magento-lts/pull/2744
* PhpStan L5 fixes for Mage/Admin by @sreichel in https://github.com/OpenMage/magento-lts/pull/2761

**Full Changelog**: https://github.com/OpenMage/magento-lts/compare/v19.4.19...v19.4.20

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v19.4.20'>v19.4.20</a>.</em>

**Comment by empiricompany** [comment]
in this release there is a typo in lib/Magento/Db/Adapter/Pdo/Mysql.php on line 135
`rray_key_exists `

```
    public function quote($value, $type = null)
    {
        $this->_connect();

        if ($type !== null
            && rray_key_exists($type = strtoupper($type), $this->_numericDataTypes)
            && $this->_numericDataTypes[$type] == Zend_Db::FLOAT_TYPE
        ) {
            $value = $this->_convertFloat($value);
            $quoteValue = sprintf('%F', $value);
            return $quoteValue;
        } elseif (is_float($value)) {
            return $this->_quote($value);
        }

        return parent::quote($value, $type);
    }
```

**Reply by fballiano** [reply]
it was already solved today (by @sreichel), one of the few directories that wasn't under phpstan check until now

**Reply by sreichel** [reply]
<s>@fballiano i know ive fixed it in one of my PRs, but its not merged. Should i open a PR for this?

edit #2788</s>

fixed in #2780

**Reply by empiricompany** [reply]
> it was already solved today (by @sreichel), one of the few directories that wasn't under phpstan check until now

Can we update this release please? Are there any other typos that need to be fixed in this release?

**Reply by fballiano** [reply]
please review https://github.com/OpenMage/magento-lts/pull/2766 and https://github.com/OpenMage/magento-lts/pull/2165, we can't release without those two.
